fn.go

```diff
@@ -269,6 +273,10 @@ func (f *Function) RunFunction(ctx context.Context, req *fnv1beta1.RunFunctionRe
 		return rsp, nil
 	}
 
+	if skipped > 0 {
+		rsp.GetDesired().GetComposite().Ready = fnv1beta1.Ready_READY_FALSE
+	}
```

Comment:
Unfortunately I don't think this'll work. Functions can't directly specify the readiness of the XR, they're supposed to influence it by specifying the readiness of composed resources.

https://github.com/crossplane/function-sdk-go/blob/29ea8ddbfa554ede602e4ca7f13c5883112159a3/proto/v1beta1/run_function.proto#L230-L231

Obviously that makes this tricky, given Crossplane wouldn't even know the composed resource exists in this case. 🤔

Reply:
Yes, tests turned out, that it is not working as expected since the ready state of the composite is hardcoded in crossplane to be determined by the ready state of the composed resources.

We have implemented a workaround that sets an arbitrary composed resource to unready until a set of dependent resources is available. Its more like a hack than an actual solution but it seems to be the only way to prevent this issue at the moment.

@negz are there any plans to change this behaviour in the implementation of the function renderer of Crossplane? It is very confusing that the ready field exists but is not recognized at all.